types/chloride/index.d.ts

```diff
@@ -0,0 +1,56 @@
+// Type definitions for chloride 2.4
+// Project: https://github.com/ssb-js/chloride
+// Definitions by: Zhiyuan Man <https://github.com/zman2013>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.4
+
+/// <reference types="node" />
+
+export interface KeyPair {
+    publicKey: Buffer;
+    secretKey: Buffer;
+}
+
+declare namespace chloride {
+    // *** hash ***
+    // sha512
+    function crypto_hash(plainText: Buffer): Buffer;
+    // sha256
+    function crypto_hash_sha256(plainText: Buffer): Buffer;
+
+    /** Signatures */
+    function crypto_sign_keypair(): KeyPair;
+    // seed's length is 24 bytes
+    function crypto_sign_seed_keypair(seed: Buffer): KeyPair;
+    // return concat( {the signature of the message}, {message} ),
+    // if just only need signature, use crypto_sign_detached instead
+    function crypto_sign(message: Buffer, secretKey: Buffer): Buffer;
+    function crypto_sign_open(signed: Buffer, publicKey: Buffer): Buffer;
+    function crypto_sign_detached(message: Buffer, secretKey: Buffer): Buffer;
+    function crypto_sign_verify_detached(signed: Buffer, message: Buffer, publicKey: Buffer): Buffer;
+
+    // *** Box ***
+    function crypto_box_keypair(): KeyPair;
+    function crypto_box_seed_keypair(seed: Buffer): KeyPair;
+    function crypto_box_easy(data: Buffer, nonce: Buffer, publicKey: Buffer, secretKey: Buffer): Buffer;
+    function crypto_box_open_easy(boxed: Buffer, nonce: Buffer, publicKey: Buffer, secretKey: Buffer): Buffer;
+
+    // *** SecretBox ***
+    function crypto_secretbox_easy(plainText: Buffer, nonce: Buffer, key: Buffer): Buffer;
+    function crypto_secretbox_open_easy(encrypted: Buffer, nonce: Buffer, key: Buffer): Buffer;
+
+    // *** random bytes ***
+    function randombytes(buf: Buffer): void;
+
+    // *** auth ***
+    function crypto_auth(data: Buffer, key: Buffer): Buffer;
+    function crypto_auth_verify(signed: Buffer, data: Buffer, key: Buffer): boolean;
+
+    // *** scalar multiplication ***
+    function crypto_scalarmult(secretKey: Buffer, publicKey: Buffer): Buffer;
+
+    // *** conversions ***
+    function crypto_sign_ed25519_sk_to_curve25519(secretKey: Buffer): Buffer;
+    function crypto_sign_ed25519_pk_to_curve25519(publicKey: Buffer): Buffer;
+}
+export default chloride;
```

Comment:
Based on the cjs examples in the repo, this looks like it should be `export =` and not `export default`; and, if I'm not mistaken, furthermore, these are all the "low level" primitives usually available under the "api" member of that? Shouldn't the high level API also be included? And usually preferred?

Reply:
1. _this looks like it should be export = and not export default_
  At last, I removed the namespace, and exported the functions instead.
2. _these are all the "low level" primitives usually available under the "api" member of that? Shouldn't the high level API also be included?_
  Actually, these are high level apis, the low level apis are from `sodium-native`. Chloride is a compatibility layer not defining the wrapping api. These apis is from difference underlaying implementations, e.g. `sodium-chloride` or `sodium-browserify-tweetnacl`. 
   Maybe you is misled by `https://www.npmjs.com/package/sodium`. After digging the code, I found the package doesn't depend on the `node-sodium`.

Reply:
Yeah, I was mislead - the examples in the `chloride` repo are of using `sodium` directly: https://github.com/ssb-js/chloride/blob/main/examples/low-level-api.js